File: apps/android-shell/app/src/main/java/com/syncclip/shell/LocalNicknameStore.kt
package com.syncclip.shell

import android.content.Context

/**
 * Optional Local Nickname stored only on this Device for UI (never on the wire).
 */
class LocalNicknameStore(context: Context) {
    private val prefs =
        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun load(): String? {
        val value = prefs.getString(KEY_NICKNAME, null)?.trim().orEmpty()
        return value.takeIf { it.isNotEmpty() }
    }

    fun save(nickname: String) {
        val trimmed = nickname.trim()
        if (trimmed.isEmpty()) {
            clear()
        } else {
            prefs.edit().putString(KEY_NICKNAME, trimmed).apply()
        }
    }

    fun clear() {
        prefs.edit().remove(KEY_NICKNAME).apply()
    }

    companion object {
        private const val KEY_NICKNAME = "local_nickname"
        private const val PREFS_NAME = "sync_clip_local_ui"
    }
}

File: apps/apple-shell/Sources/AppleShellCore/LocalNicknameStore.swift
import Foundation

/// Optional Local Nickname stored only on this Device for UI (never on the wire).
public protocol LocalNicknameStoring: AnyObject {
    func clear()
    func load() -> String?
    func save(_ nickname: String)
}

public final class UserDefaultsLocalNicknameStore: LocalNicknameStoring {
    private let defaults: UserDefaults
    private let key: String

    public init(
        defaults: UserDefaults = .standard,
        key: String = "com.syncclip.shell.localNickname"
    ) {
        self.defaults = defaults
        self.key = key
    }

    public func load() -> String? {
        let value = defaults.string(forKey: key)?.trimmingCharacters(in: .whitespacesAndNewlines)
        guard let value, !value.isEmpty else { return nil }
        return value
    }

    public func save(_ nickname: String) {
        let trimmed = nickname.trimmingCharacters(in: .whitespacesAndNewlines)
        if trimmed.isEmpty {
            clear()
        } else {
            defaults.set(trimmed, forKey: key)
        }
    }

    public func clear() {
        defaults.removeObject(forKey: key)
    }
}
